internal/runners/activate/default.go

```diff
@@ -0,0 +1,269 @@
+package activate
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	rt "runtime"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/strutils"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/gobuffalo/packr"
+	"github.com/thoas/go-funk"
+)
+
+type DefaultConfigurer interface {
+	Set(key string, value interface{})
+	GetString(key string) string
+}
+
+type exeFile struct {
+	fpath string
+	name  string
+	ext   string
+}
+
+func getDefaultProjectPath(d DefaultConfigurer) string {
+	return d.GetString("default_project_path")
+}
+
+func setDefaultProjectPath(d DefaultConfigurer, path string) {
+	d.Set("default_project_path", path)
+}
+
+type primable interface {
+	primer.Outputer
+}
+
+// gets the intended target for a shim
+func shimTarget(fn string) (string, error) {
+	contents, err := ioutil.ReadFile(fn)
+	if err != nil {
+		return "", err
+	}
+
+	targetRe := regexp.MustCompile("^ target: (.)$")
+	target := targetRe.FindString(string(contents))
+	if target == "" {
+		return "", errs.New("Target file is not a shim.")
+	}
+
+	return target, nil
+}
+
+// rollbackShims removes all shims in the global binary directory that target a previous default project
+// If the target of a shim does not exist anymore, the shim is also removed.
+func rollbackShims(cfg DefaultConfigurer) error {
+	projectPath := getDefaultProjectPath(cfg)
+	binDir := config.GlobalBinPath()
+
+	// remove symlinks pointing to default project
+	files, err := ioutil.ReadDir(binDir)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		fn := f.Name()
+		target, err := shimTarget(fn)
+		if err != nil {
```

Comment:
Okay, I'm getting quite confused now.  Should the implementation allow to define parallel default projects?  Because I thought that you could only ever have one project be the default one ie., Python or Perl.

I added this logic and the `shimTarget` function to ensure that we are not deleting too much, but above I argued that it is probably cleaner for this iteration to just check if the file is probably a shim, and if that is the case, I remove it.  That is in line with your first comment here.  Would that still be okay?

Reply:
Yes it should allow parallel defaults for different languages. You are setting the default for the language of the project, not the default for any project ever.

I had talked about this while pointing and it's somewhat implied in the ACs, guess I could've made it more explicit.

Reply:
Yes, I remember it now.  Sorry, I forgot, and did not infer it from the ACs.